### ISSUE TITLE
Search context should be changed with header links

### ISSUE THREAD

**Issue by tofumatt** [mozilla/addons-frontend#2513]
### Describe the problem and steps to reproduce it:
If a search term is entered in "Explore" mode, clicking "Extensions" should continue the search with `type=extension` added to the search. Same for themes, or switching back to explore.

Depends on #2468.

Realistically, doesn't make sense to do until we implement #2737 and move more search data into state and stop depending on location/state together as we do now.

**Comment by muffinresearch** [comment]
@tofumatt This looks to be complete - however when starting on the homepage and searching you get urls like:

https://addons-dev.allizom.org/en-GB/firefox/search/?q=foo&type

I think that should just be: 

https://addons-dev.allizom.org/en-GB/firefox/search/?q=foo

**Comment by tofumatt** [comment]
Ah yes, I noticed that awhile back and forgot to correct it. I wonder if it's a change somewhere in our libraries... odd the tests didn't catch it as I thought they compared exact URLs...

I'll file a new issue for that as it gets ignored by the API it seems, so it doesn't affect much... but would be good to fix 😄

**Comment by tofumatt** [comment]
Oh, also it's not complete because clicking on one of those links doesn't continue a search but change the type.

**Comment by muffinresearch** [comment]
> Oh, also it's not complete because clicking on one of those links doesn't continue a search but change the type.

Ah ok, makes sense.

**Comment by tofumatt** [comment]
I was thinking about this more and it's a weird one because we only want the link to change the search context when the user is already searching... we can change the links in the header using the current search state but we'd need something like `react-router-redux` to know that we're *on* a search URL.

**Comment by muffinresearch** [comment]
Chatted about this and we decided that we wouldn't fix it.
